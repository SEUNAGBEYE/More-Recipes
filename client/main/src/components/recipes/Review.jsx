import React from 'react';
import PropTypes from 'prop-types';

const Review = (props) => (
  <div className="row" style={{ marginBottom: 15 }}>
    <div className="" style={{
      backgroundColor: '#f8f9fa',
      border: '1 solid #f8f9fa',
      width: '100vw'
    }}
    >
      <div className="col-md-1 review" style={{ paddingTop: 10 }}>
        <img style={{ width: 30, height: 30, borderRadius: 80 }}
          src={props.review.user.profilePicture}/>
      </div>
      <div className="col-md-11 review">
        <h5 style={{ color: 'orange', marginTop: 5 }}>{
          props.review.user.firstName}
        </h5>
        {props.review.body}
      </div>
    </div>
  </div>
);

const propTypes = {
  review: PropTypes.object.isRequired
};

Review.propTypes = propTypes;
export default Review;
