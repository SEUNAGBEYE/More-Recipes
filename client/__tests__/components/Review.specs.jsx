import React from 'react';
import { shallow } from 'enzyme';
import Review from '../../main/src/components/recipes/Review';

const props = {
  review: {
    user: {
      profilePicture: 'This is my profile picture',
      firstName: 'Seun'
    },
    body: 'Ths is my review'
  }
};
  

describe('# Review', () => {
  const wrapper = shallow(
      <Review {...props} />
  );
  it('should render successfully', (done) => {
    expect(wrapper).toBeDefined();
    expect(wrapper.getElement().type).toBe('div');
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
