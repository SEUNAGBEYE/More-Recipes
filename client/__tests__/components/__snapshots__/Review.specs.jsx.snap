// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Review should render successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Review
    review={
      Object {
        "body": "Ths is my review",
        "user": Object {
          "firstName": "Seun",
          "profilePicture": "This is my profile picture",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className=""
        style={
          Object {
            "backgroundColor": "#f8f9fa",
            "border": "1 solid #f8f9fa",
            "width": "100vw",
          }
        }
      >
        <div
          className="col-md-1 review"
          style={
            Object {
              "paddingTop": 10,
            }
          }
        >
          <img
            src="This is my profile picture"
            style={
              Object {
                "borderRadius": 80,
                "height": 30,
                "width": 30,
              }
            }
          />
        </div>
        <div
          className="col-md-11 review"
        >
          <h5
            style={
              Object {
                "color": "orange",
                "marginTop": 5,
              }
            }
          >
            Seun
          </h5>
          Ths is my review
        </div>
      </div>,
      "className": "row",
      "style": Object {
        "marginBottom": 15,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-md-1 review"
            style={
              Object {
                "paddingTop": 10,
              }
            }
          >
            <img
              src="This is my profile picture"
              style={
                Object {
                  "borderRadius": 80,
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </div>,
          <div
            className="col-md-11 review"
          >
            <h5
              style={
                Object {
                  "color": "orange",
                  "marginTop": 5,
                }
              }
            >
              Seun
            </h5>
            Ths is my review
          </div>,
        ],
        "className": "",
        "style": Object {
          "backgroundColor": "#f8f9fa",
          "border": "1 solid #f8f9fa",
          "width": "100vw",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              src="This is my profile picture"
              style={
                Object {
                  "borderRadius": 80,
                  "height": 30,
                  "width": 30,
                }
              }
            />,
            "className": "col-md-1 review",
            "style": Object {
              "paddingTop": 10,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "This is my profile picture",
              "style": Object {
                "borderRadius": 80,
                "height": 30,
                "width": 30,
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h5
                style={
                  Object {
                    "color": "orange",
                    "marginTop": 5,
                  }
                }
              >
                Seun
              </h5>,
              "Ths is my review",
            ],
            "className": "col-md-11 review",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Seun",
                "style": Object {
                  "color": "orange",
                  "marginTop": 5,
                },
              },
              "ref": null,
              "rendered": "Seun",
              "type": "h5",
            },
            "Ths is my review",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className=""
          style={
            Object {
              "backgroundColor": "#f8f9fa",
              "border": "1 solid #f8f9fa",
              "width": "100vw",
            }
          }
        >
          <div
            className="col-md-1 review"
            style={
              Object {
                "paddingTop": 10,
              }
            }
          >
            <img
              src="This is my profile picture"
              style={
                Object {
                  "borderRadius": 80,
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </div>
          <div
            className="col-md-11 review"
          >
            <h5
              style={
                Object {
                  "color": "orange",
                  "marginTop": 5,
                }
              }
            >
              Seun
            </h5>
            Ths is my review
          </div>
        </div>,
        "className": "row",
        "style": Object {
          "marginBottom": 15,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-md-1 review"
              style={
                Object {
                  "paddingTop": 10,
                }
              }
            >
              <img
                src="This is my profile picture"
                style={
                  Object {
                    "borderRadius": 80,
                    "height": 30,
                    "width": 30,
                  }
                }
              />
            </div>,
            <div
              className="col-md-11 review"
            >
              <h5
                style={
                  Object {
                    "color": "orange",
                    "marginTop": 5,
                  }
                }
              >
                Seun
              </h5>
              Ths is my review
            </div>,
          ],
          "className": "",
          "style": Object {
            "backgroundColor": "#f8f9fa",
            "border": "1 solid #f8f9fa",
            "width": "100vw",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                src="This is my profile picture"
                style={
                  Object {
                    "borderRadius": 80,
                    "height": 30,
                    "width": 30,
                  }
                }
              />,
              "className": "col-md-1 review",
              "style": Object {
                "paddingTop": 10,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "src": "This is my profile picture",
                "style": Object {
                  "borderRadius": 80,
                  "height": 30,
                  "width": 30,
                },
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h5
                  style={
                    Object {
                      "color": "orange",
                      "marginTop": 5,
                    }
                  }
                >
                  Seun
                </h5>,
                "Ths is my review",
              ],
              "className": "col-md-11 review",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Seun",
                  "style": Object {
                    "color": "orange",
                    "marginTop": 5,
                  },
                },
                "ref": null,
                "rendered": "Seun",
                "type": "h5",
              },
              "Ths is my review",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
